src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1195,7 +1199,8 @@ private void ParseModifiers(SyntaxListBuilder tokens, bool forAccessors)
                             // Misplaced partial
                             // TODO(https://github.com/dotnet/roslyn/issues/22439):
                             // We should consider moving this check into binding, but avoid holding on to trees
```

Comment:
@agocke I had to move this check to binding (by removing it here). I'm not sure what refactoring you had in mind.